libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -52,12 +71,48 @@ getDState :: NewEpochState era -> DState era
 getDState = certDState . lsCertState . esLState . nesEs
 
 queryConstitution :: EraGov era => NewEpochState era -> Maybe (Constitution era)
-queryConstitution nes =
-  getConstitution (nes ^. nesEpochStateL . esLStateL . lsUTxOStateL . utxosGovStateL)
+queryConstitution = getConstitution . queryGovState
 
 queryConstitutionHash ::
   EraGov era =>
   NewEpochState era ->
   Maybe (SafeHash (EraCrypto era) AnchorData)
 queryConstitutionHash nes =
   anchorDataHash . constitutionAnchor <$> queryConstitution nes
+
+-- | This query returns all of the state related to governance
+queryGovState :: NewEpochState era -> GovState era
+queryGovState nes = nes ^. nesEpochStateL . esLStateL . lsUTxOStateL . utxosGovStateL
+
+-- | Query DRep state.
+queryDRepState ::
+  NewEpochState era ->
+  -- | Specify a set of DRep credentials whose state should be returned. When this set is
+  -- empty, states for all of the DReps will be returned.
+  Set (Credential 'DRepRole (EraCrypto era)) ->
+  Map (Credential 'DRepRole (EraCrypto era)) (DRepState (EraCrypto era))
+queryDRepState nes creds
+  | null creds = drepsState
+  | otherwise = drepsState `Map.restrictKeys` creds
+  where
+    drepsState = vsDReps $ certVState $ lsCertState $ esLState $ nesEs nes
```

Comment:
A little inconsistent with the above queryGovState which uses lenses, so was expecting:

 drepsState = nes ^. nesEsL . esLStateL . lsCertStateL . certVStateL . vsDRepsL

But it's maybe overkill to use lenses if the code is almost identical without.

Reply:
I don't think we should worry about lens consistency for now. I have a good idea how we can implement a consistent way of dealing with ledger state with a nice lens interface. But that will come later, definitely after SanchoNet.